packages/core/redux/wallets/actions.js

```diff
@@ -64,74 +69,37 @@ export const initWallets = () => (dispatch) => {
 }
 
 const initWalletsFromKeys = () => async (dispatch, getState) => {
-  const state = getState()
-  const account = getPersistAccount(state)
-  const wallets = []
-
-  const ethereumSigner = getEthereumSigner(state)
-  const ethAddress = await ethereumSigner.getAddress()
-
-  wallets.push(new WalletModel({
-    address: ethAddress,
-    blockchain: BLOCKCHAIN_ETHEREUM,
-    isMain: true,
-    walletDerivedPath: account.decryptedWallet.entry.encrypted[0].path,
-  }))
-
-  const bitcoinSigner = getBitcoinSigner(state)
-  if (bitcoinSigner) {
-    wallets.push(new WalletModel({
-      address: bitcoinSigner.getAddress(),
-      blockchain: BLOCKCHAIN_BITCOIN,
-      isMain: true,
-      walletDerivedPath: account.decryptedWallet.entry.encrypted[0].path,
-    }))
-  }
-
-  const bitcoinCashSigner = getBitcoinCashSigner(state)
-  if (bitcoinCashSigner) {
-    wallets.push(new WalletModel({
-      address: bitcoinCashSigner.getAddress(),
-      blockchain: BLOCKCHAIN_BITCOIN_CASH,
-      isMain: true,
-      walletDerivedPath: account.decryptedWallet.entry.encrypted[0].path,
-    }))
-  }
-
-  const litecoinSigner = getLitecoinSigner(state)
-  if (litecoinSigner) {
-    wallets.push(new WalletModel({
-      address: litecoinSigner.getAddress(),
-      blockchain: BLOCKCHAIN_LITECOIN,
-      isMain: true,
-      walletDerivedPath: account.decryptedWallet.entry.encrypted[0].path,
-    }))
-  }
-
-  const nemSigner = getNemSigner(state)
-  if (nemSigner) {
-    wallets.push(new WalletModel({
-      address: nemSigner.getAddress(),
-      blockchain: BLOCKCHAIN_NEM,
-      isMain: true,
-      walletDerivedPath: account.decryptedWallet.entry.encrypted[0].path,
-    }))
-  }
-
-  const wavesSigner = getWavesSigner(state)
-  if (wavesSigner) {
-    wallets.push(new WalletModel({
-      address: wavesSigner.getAddress(),
-      blockchain: BLOCKCHAIN_WAVES,
-      isMain: true,
-      walletDerivedPath: account.decryptedWallet.entry.encrypted[0].path,
-    }))
-  }
+  const state = getState();
+  const account = getPersistAccount(state);
+  const wallets = [];
+
+  await [
+    BLOCKCHAIN_BITCOIN,
+    BLOCKCHAIN_BITCOIN_CASH,
+    BLOCKCHAIN_DASH,
+    BLOCKCHAIN_LITECOIN,
+    BLOCKCHAIN_ETHEREUM,
+    BLOCKCHAIN_NEM,
+    BLOCKCHAIN_WAVES
+  ].map(async (blockchainType) => {
+    const signer = getSigner(blockchainType, state);
+
+    if(signer !== undefined) {
+      const address = blockchainType === BLOCKCHAIN_ETHEREUM ? (await signer.getAddress()) : signer.getAddress();
```

Comment:
Why do you use await only with EHTEREUM? LedgerDevice is also have async getAddress method

Reply:
Because in the original source code `await` was before `ethereumSigner` only (see https://github.com/ChronoBank/ChronoMint/blob/develop/packages/core/redux/wallets/actions.js#L72). I did not look into classes I did not modify. I just rewrite the source code in a shorter manner keeping it working as it was working (except several obvious mistakes). Now I see that it would be better to use await for all cases.